torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -224,6 +224,8 @@ def _share_state_and_init_handle_attrs(
         handle._has_optim_in_backward = flat_param._params is not None and any(
             hasattr(param, "_in_backward_optimizers") for param in flat_param._params
         )
+        if handle._has_optim_in_backward:
+            torch._C._log_api_usage_once("fsdp.optimizer_in_backward")
```

Comment:
Do you want to log one usage per FSDP instance / flat parameter (as opposed to once per model)?

Reply:
I think this will only log once per application, see https://fburl.com/code/kp0ktnr6. Basically even if this is called for every FlatParam, the string is cached so it only gets logged once. I can still move it to a better location if we're concerned about the overhead.

Reply:
Ah, I see. It is not a big deal to change, but maybe doing an `any()` over the handles and trying to log it once might make it more obviously logged once.